flycheck.el

```diff
@@ -8711,6 +8715,25 @@ rustc command.  See URL `https://www.rust-lang.org'."
                (and (flycheck-buffer-saved-p)
                     (locate-dominating-file (buffer-file-name) "Cargo.toml"))))
 
+(flycheck-define-checker rust-cargo-clippy
+  "A Rust syntax checker and linter using clippy.
+
+This syntax checker needs the cargo clippy subcommand. See URL `https://github.com/Manishearth/rust-clippy'"
+  :command ("cargo" "clippy"
+            (eval flycheck-cargo-clippy-args)
+            "--"  "--error-format=json"
+            (option-list "-L" flycheck-rust-library-path concat)
+            (eval flycheck-rust-args))
+  :error-parser flycheck-parse-rust
+  :error-explainer flycheck-rust-error-explainer
+  :modes rust-mode
+  :predicate (lambda ()
+               ;; Since we build the entire project with cargo clippy we require
+               ;; that the buffer is saved.  And of course, we also need a Cargo
+               ;; file :)
+               (and (flycheck-buffer-saved-p)
+                    (locate-dominating-file (buffer-file-name) "Cargo.toml"))))
```

Comment:
Please move the check for `Cargo.toml` into an `:enabled` function (see `javascript-eslint` for an example of the `:enabled` field), so that we only check for the file once when the checker is used first and not every single time.

I presume the `clippy` command is not built-in to `cargo`?  If so `:enabled` also needs to check whether the subcommand is present.